src/py42/services/_connection.py

```diff
@@ -230,6 +229,8 @@ def _prepare_request(
         headers.update(self._headers)
         if data and u"Content-Type" not in headers:
             headers.update({u"Content-Type": u"application/json"})
+        if u"Accept" not in headers:
```

Comment:
it's starting to look like these updates are being applied in the wrong order -- we shouldn't have to make an exception here for every header that a user passes in. I probably gave some incorrect guidance on this whole thing earlier, sorry about that.

I'm thinking what we want is something that ensure that `headers` gets applied last so that no matter what, whatever the function submitted would take precedence, like:

```python
final_headers = {}
final_headers.update(self._headers)
headers = _create_user_headers(headers)
final_headers.update(headers)
request = Request(... headers=final_headers)
```